web/src/engine/osk/src/banner/suggestionBanner.ts

```diff
@@ -585,6 +596,30 @@ export class SuggestionBanner extends Banner {
       }
 
       const terminationHandler = () => {
+        const currentSuggestions = this.currentSuggestions;
+        // First, schedule reselection of the autoselected suggestion.
+        // We shouldn't do it synchronously, as suggestion acceptance triggers
+        // _after_ this handler is called.
+        // Delaying via the task queue is enough to get the desired order of events.
+        timedPromise(0).then(async () => {
+          // If the suggestion list instance has changed, our state has changed; do
+          // not reselect.
+          if(currentSuggestions != this.currentSuggestions) {
+            return;
+          }
+
+          // The suggestions are still current?  Then restore the original
+          // auto-correct suggestion and its highlighting.
+          this._predictionContext.selected = autoselection;
+          if(autoselection) {
+            this.options.forEach((entry) => {
+              if(entry.suggestion == autoselection) {
```

Comment:
Can there be more than one `entry.suggestion == autoselection`? Otherwise we could stop the loop  after we did highlight the entry.